src/VirtoCommerce.CustomerSegmentsModule.Core/Models/CustomerSegmentTree.cs

```diff
@@ -1,12 +1,38 @@
+using System.Collections.Generic;
 using VirtoCommerce.CoreModule.Core.Conditions;
+using VirtoCommerce.CustomerModule.Core.Model.Search;
+using VirtoCommerce.Platform.Core.Common;
 
 namespace VirtoCommerce.CustomerSegmentsModule.Core.Models
 {
-    public class CustomerSegmentTree : BlockConditionAndOr
+    public class CustomerSegmentTree : BlockConditionAndOr, ICanBuildSearchCriteria
     {
         public CustomerSegmentTree()
         {
             All = true;
         }
+
+        public virtual MembersSearchCriteria BuildSearchCriteria(IMemberSearchCriteriaBuilder builder)
+        {
+            if (!Children.IsNullOrEmpty())
+            {
+                BuildSearchCriteria(Children, builder);
+            }
+
+            return builder.Build();
+        }
+
+        protected virtual void BuildSearchCriteria(IList<IConditionTree> children, IMemberSearchCriteriaBuilder builder)
```

Comment:
Rename this method to give more clearance what this method does. Maybe "EvaluateSearchCriteria"?